Adds a positive moodlet to seeing a dog playing

stinky-lizard opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds a small positive moodlet (intensity = 2, timeout = 3 minutes) whenever one sees a dog playing ("chases its tail", "dances around"). For instance: "Ian dances around!" -> "Aww, Ian's having fun!"

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

It's cute. Personally I feel happy whenever my pet is having fun. Also, another positive moodlet (however small) will offset the chaos on station, and dogs aren't all that common on station. There's usually only, what, 3? And they're usually dead, so they can't play.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: You now get a positive moodlet whenever you see a dog playing.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

stinky-lizard committed:
this seems to work in game

im doing this from github bc i fucked up my local repo

stinky-lizard committed:
this seems to work in game pt 2

im doing this from github bc i fucked up my local repo

stinky-lizard committed:
this seems to work in game part 3: electric bee

im doing this on github bc i fucked up my local repo. that's why there's three commits of single files instead of one with 3 files

Archanial commented on code/datums/ai/dog/dog_controller.dm:
```diff
@@ -75,6 +75,9 @@
 		living_pawn.manual_emote(pick("dances around.", "chases [living_pawn.p_their()] tail!</span>"))
 		living_pawn.AddComponent(/datum/component/spinny)
+		for (var/mob/i in oviewers(living_pawn))
+			if (ishuman(i) && i.client) //if viewer is human and a player. maybe check if isn't SSD? also, do non-human players have moods?
```
check for mind not client

stinky-lizard replied:
Oh, okay. I'm new to SS13 code, so I didn't know about mind. Does the new commit work?

Archanial replied:
Yes

stinky-lizard committed:
changed check for mob.client to mob.mind

stinky-lizard committed:
i.mind again.
slightly changed & trimmed extraneous comments, updated dog.dm as well

Archanial reviewed (dismissed):
just so checks run

Archanial commented on code/datums/ai/dog/dog_controller.dm:
```diff
@@ -75,6 +75,9 @@
 		living_pawn.manual_emote(pick("dances around.", "chases [living_pawn.p_their()] tail!</span>"))
 		living_pawn.AddComponent(/datum/component/spinny)
+		for (var/mob/i in oviewers(living_pawn))
+			if (ishuman(i) && i.mind) //if viewer is human and a player. do non-human players have moods?
```
```suggestion
		for(var/mob/living/carbon/human/H in oviewers(living_pawn))
			if(H.mind)
```
All mob/living can have mind, what you're doing here is giving the moodlet to only humans
Oh and that for look has istype built into it

stinky-lizard committed:
Update code/datums/ai/dog/dog_controller.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

stinky-lizard committed:
Update dog_controller.dm

stinky-lizard committed:
removed changes from dog.dm to keep it simpler
keeping the mood signals in dog.dm when they aren't used is just
more complexity for no benefit. it's easier to just drop them
